catalyst-toolbox-cli/src/cli/recovery/tally/fragments.rs

```diff
@@ -0,0 +1,94 @@
+use std::fs;
+use std::io;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+
+use bincode::Options;
+use jormungandr_lib::interfaces::PersistentFragmentLog;
+
+struct FileFragments {
+    reader: BufReader<fs::File>,
+}
+
+struct FileFragmentsIterator {
+    reader: BufReader<fs::File>,
+}
+
+impl FileFragments {
+    pub fn from_path(file_path: PathBuf) -> std::io::Result<Self> {
+        fs::File::open(file_path).map(|file| Self {
+            reader: BufReader::new(file),
+        })
+    }
+}
+
+impl IntoIterator for FileFragments {
+    type Item = PersistentFragmentLog;
+    type IntoIter = FileFragmentsIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        FileFragmentsIterator {
+            reader: self.reader,
+        }
+    }
+}
+
+impl Iterator for FileFragmentsIterator {
+    type Item = PersistentFragmentLog;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let codec = bincode::DefaultOptions::new()
+            .with_fixint_encoding()
+            .allow_trailing_bytes();
+        codec.deserialize_from(&mut self.reader).ok()
+    }
+}
+
+pub fn get_fragments_log_files_path(folder: &Path) -> io::Result<impl Iterator<Item = PathBuf>> {
+    let mut entries: Vec<_> = fs::read_dir(folder)?
```

Comment:
A similar note here: I would rather write a `for` loop and return errors from `entry`. I am not even sure when they could appear, probably a permissions problem which is very unlikely in this setup.